lib/lightdata.h

```diff
@@ -12,17 +12,58 @@ class lightdata {
   unsigned char fine = 0;
   unsigned char tdc = 0;
 
-  int chip() const { return index / 32; };
-  int eoch() const { return index % 64; };
-  int cindex() const { return tdc + 4 * index; };
-  
   lightdata(unsigned char _device, unsigned char _index, unsigned char _coarse, unsigned char _fine, unsigned char _tdc) :
     device(_device),
     index(_index),
     coarse(_coarse),
     fine(_fine),
     tdc(_tdc) { };
-    
+
+  int chip() const { return index / 32; };
+  int eoch() const { return index % 64; };
+  int cindex() const { return tdc + 4 * index; };
+  float time() const;
+
+  /** calibration **/
+
+  static float fine_if[16][768];
+  static float fine_cut[16][768];
+  static float fine_off[16][768];
+  static bool load_fine_calibration(std::string filename);
+
 };
 
+float lightdata::fine_if[16][768] = {0.};
+float lightdata::fine_cut[16][768] = {0.};
+float lightdata::fine_off[16][768] = {0.};
+
+float
+lightdata::time() const
+{
+  auto ci = cindex();
+  float corr = (float)fine * fine_if[device][ci] + fine_off[device][ci];
+  if (fine > fine_cut[device][ci]) corr -= 1.;
+  auto time = (float)coarse - corr;
+  return time;
+}
+
+bool
+lightdata::load_fine_calibration(std::string filename)
```

Comment:
Do we foresee to have multiple files for calibration? Like one file for device X and another for device Y? Because then this will overwrite the previously loaded files

Reply:
Moreover: do we foresee to work on multiple runs simultaneously with different calibrations?

Reply:
only one calibration file for run